inference-engine/src/vpu/common/src/ngraph/transformations/dynamic_to_static_shape_split.cpp

```diff
@@ -0,0 +1,60 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "vpu/ngraph/transformations/dynamic_to_static_shape_broadcast.hpp"
+
+#include "vpu/ngraph/operations/dynamic_shape_resolver.hpp"
+#include "vpu/ngraph/utilities.hpp"
+#include "vpu/utils/error.hpp"
+
+#include "ngraph/graph_util.hpp"
+#include "ngraph/opsets/opset5.hpp"
+#include "ngraph/validation_util.hpp"
+
+namespace vpu {
+
+void dynamicToStaticShapeSplit(std::shared_ptr<ngraph::Node> target) {
+    const auto split = ngraph::as_type_ptr<ngraph::opset5::Split>(target);
+    VPU_THROW_UNLESS(split,
+                     "dynamicToStaticShapeSplit transformation is not applicable for {}, "
+                     "it should be {} instead",
+                     target, ngraph::opset5::Split::type_info);
+
+    const auto numSplits = split->get_num_splits();
+
+    const auto dsr = ngraph::as_type_ptr<ngraph::vpu::op::DynamicShapeResolver>(target->input_value(0).get_node_shared_ptr());
+    VPU_THROW_UNLESS(dsr, "DynamicToStaticShape transformation for {} of type {} expects {} as input with index {}",
+                     target->get_friendly_name(), target->get_type_info(), ngraph::vpu::op::DynamicShapeResolver::type_info, 0);
+
+    const auto dataShape = dsr->input_value(1).get_node_shared_ptr();
+    const auto dataShapeType = dataShape->get_element_type();
+    const auto axisNode = ngraph::as_type_ptr<ngraph::opset5::Constant>(target->input_value(1).get_node_shared_ptr());
+    VPU_THROW_UNLESS(axisNode, "dynamicToStaticShapeSplit transformation is not applicable for {}, dynamic axis is not supported", target);
+
+    const auto dataRank = target->get_input_partial_shape(0).rank();
+    VPU_THROW_UNLESS(dataRank.is_static(), "dynamicToStaticShapeSplit transformation for {} doesn't support dynamic rank", target);
+    int64_t axis = ngraph::normalize_axis(target->description(), axisNode->cast_vector<int64_t>()[0], dataRank);
```

Comment:
I'd add assert on size of `axisNode->cast_vector<int64_t>()`

Reply:
Good point! Done.